diff --git a/core/modules/field/src/Entity/FieldConfig.php b/core/modules/field/src/Entity/FieldConfig.php
index 8109d138da..3bd5ee0654 100644
--- a/core/modules/field/src/Entity/FieldConfig.php
+++ b/core/modules/field/src/Entity/FieldConfig.php
@@ -369,7 +369,7 @@ public function getUniqueIdentifier() {
    * @param string $field_name
    *   Name of the field.
    *
-   * @return Drupal\field\FieldConfigInterface|null
+   * @return \Drupal\field\FieldConfigInterface|null
    *   The field config entity if one exists for the provided field
    *   name, otherwise NULL.
    */
