diff --git a/src/ChromePage.php b/src/ChromePage.php
index 2d65200e53ea3744d4bab8ea460f97971ef2cff0..399d4ce8bc83602fb6d9e32abc3542c87aa257bf 100644
--- a/src/ChromePage.php
+++ b/src/ChromePage.php
@@ -172,7 +172,7 @@ class ChromePage extends DevToolsConnection
                     }
                     break;
                 default:
-                    continue;
+                    break;
             }
         }
